tempesta_fw/http_parser.c

```diff
@@ -1375,6 +1395,338 @@ __parse_content_type(TfwHttpMsg *hm, unsigned char *data, size_t len)
 	return r;
 }
 
+static int
+__strdup_multipart_boundaries(TfwHttpReq *req)
+{
+	unsigned char *data_raw, *data, *ptr_raw, *ptr;
+	TfwStr *c, *end;
+
+	data_raw = tfw_pool_alloc(req->pool, req->multipart_boundary_raw.len);
+	data = tfw_pool_alloc(req->pool, req->multipart_boundary.len);
+
+	if (!data_raw || !data)
+		return -ENOMEM;
+
+	ptr_raw = data_raw;
+	ptr = data;
+	TFW_STR_FOR_EACH_CHUNK(c, &req->multipart_boundary_raw, end) {
+		memcpy_fast(ptr_raw, c->ptr, c->len);
+		ptr_raw += c->len;
+		if (c->flags & TFW_STR_VALUE) {
+			memcpy_fast(ptr, c->ptr, c->len);
+			ptr += c->len;
+		}
+	}
+
+	BUG_ON(ptr_raw != data_raw + req->multipart_boundary_raw.len);
```

Comment:
Better to use TFW_WARN here and return error code to the caller. If `__req_parse_content_type()` has a bug, the currently parsed request will be dropped, bu the server will remain alive.

Reply:
Agreed, changed to use TFW_WARN. I haven't added any information about lengths, as it seem to be insufficient for debugging anyway.